web-components/src/components/table-advanced/TableAdvanced.ts

```diff
@@ -68,20 +71,39 @@ export namespace TableAdvanced {
     connectedCallback() {
       super.connectedCallback();
       document.addEventListener("dragover", this.dragover);
+      this.updateTableConfig();
+      this.updateTableData();
       this.populateTable();
     }
 
     protected update(changedProperties: PropertyValues) {
       super.update(changedProperties);
-      if (changedProperties.has("data")) {  
-         this.updateDataInTable();
+      if (changedProperties.has("data")) {
+        this.updateTableData();
+        this.updateDataInTable();
+      }
+    }
+
+    private updateTableData() {
+      if (typeof this.data === "string") {
+        this.tableData = JSON.parse(this.data);
```

Comment:
If there is an invalid format the JSON.parse throws error, but its not very readbale. add try catch and throw parse error along with readable message

Reply:
Will incorporate the try and catch